bolt/src/main/java/com/slack/api/bolt/App.java

```diff
@@ -634,15 +653,15 @@ public App message(String pattern, BoltEventHandler<MessageEvent> messageHandler
     }
 
     public App message(Pattern pattern, BoltEventHandler<MessageEvent> messageHandler) {
-        event(MessageEvent.class, (event, ctx) -> {
+        event(MessageEvent.class, true, (event, ctx) -> {
             String text = event.getEvent().getText();
             if (log.isDebugEnabled()) {
                 log.debug("Run a message event handler (pattern: {}, text: {})", pattern, text);
             }
             if (text != null && pattern.matcher(text).matches()) {
                 return messageHandler.apply(event, ctx);
             } else {
-                return ctx.ack();
+                return null;
```

Comment:
This is a behavior change for unmatched patterns but I would call it a bug fix rather than a breaking change. If the message event handler does not match, bolt-java should not acknowledge the request. This is a major difference from bolt-js in its design. If we receive more feedback on the consistent behavior with bolt-js in the future, it's relatively easy to introduce a new option to turn on "auto-acknowledgement" for all Events API requests like bolt-js does. However, this is out of this PR's scope.